src/linclust/kmermatcher.h

```diff
@@ -205,34 +232,34 @@ size_t queueNextEntry(KmerPositionQueue &queue, int file, size_t offsetPos, T *e
 
 void setKmerLengthAndAlphabet(Parameters &parameters, size_t aaDbSize, int seqType);
 
-template <int TYPE, typename T, typename seqLenType>
-void writeKmersToDisk(std::string tmpFile, KmerPosition<seqLenType> *kmers, size_t totalKmers);
+template <int TYPE, typename T, typename seqLenType, bool IncludeAdjacentSeq>
+void writeKmersToDisk(std::string tmpFile, KmerPosition<seqLenType, IncludeAdjacentSeq> *kmers, size_t totalKmers);
 
-template <int TYPE, typename T>
-void writeKmerMatcherResult(DBWriter & dbw, KmerPosition<T> *hashSeqPair, size_t totalKmers,
+template <int TYPE, typename T, bool IncludeAdjacentSeq>
+void writeKmerMatcherResult(DBWriter & dbw, KmerPosition<T, IncludeAdjacentSeq> *hashSeqPair, size_t totalKmers,
                             std::vector<char> &repSequence, size_t threads);
 
 
-template <typename T>
-KmerPosition<T> * doComputation(size_t totalKmers, size_t split, size_t splits, std::string splitFile,
+template <typename T, bool IncludeAdjacentSeq>
+KmerPosition<T, IncludeAdjacentSeq> * doComputation(size_t &totalKmers, size_t split, size_t splits, std::string splitFile,
                                 DBReader<unsigned int> & seqDbr, Parameters & par, BaseMatrix  * subMat,
                                 size_t KMER_SIZE, size_t chooseTopKmer, float chooseTopKmerScale = 0.0);
-template <typename T>
-KmerPosition<T> *initKmerPositionMemory(size_t size);
+template <typename T, bool IncludeAdjacentSeq = false>
```

Comment:
defined default value for methods that are used elsewhere than in `kmermatcher` (i.e. `kmersearch`, `kmerindexdb`)